ibridges/irodsconnector/data_operations.py

```diff
@@ -110,56 +120,157 @@ def irods_put(self, local_path: str, irods_path: str, resc_name: str = ''):
+        ----------
+        local_path : Path
+            Absolute path to the directory to upload
+        irods_path : IrodsPath
+            Absolute irods destination path
+        resc_name : str
+            Name of the resource to which data is uploaded, by default the server will decide
+        options : dict
+            More options for the upload
+        """
+        # get all files and their relative path to local_path
+        if not local_path.is_dir():
+            raise ValueError("local_path must be a directory.")
+
+        files = []  # (rel_path, name)
+
+        for w in os.walk(local_path):
+            files.extend([(w[0].removeprefix(str(local_path)), f) for f in w[2]])
+
+        upload_path = IrodsPath(self.session,
+                                str(irods_path.joinpath(local_path.name)))
+        _ = self.create_collection(upload_path)
+
+        for f in files:
+            # ensure iRODS collection exists
+            _ = self.create_collection(upload_path.joinpath(f[0].lstrip('/')))
+            # call irods put for Path(local_path, f[0], f[1]) to destination collection
+            dest = IrodsPath(self.session, str(upload_path), f[0].lstrip('/'))
+            self.irods_put(Path(str(local_path), f[0].lstrip('/'), f[1]), dest,
+                           resc_name, options)
+
+    def download_collection(self, irods_path: IrodsPath, local_path: Path,
+                            overwrite: bool = False, options: Optional[dict] = None):
+        """Download a collection to the local filesystem
+
+        Parameters
+        ----------
+        irods_path : IrodsPath
+            Absolute irods source path pointing to a collection
+        local_path : Path
+            Absolute path to the destination directory
+        overwrite : bool
+            Overwrite existing local data
+        options : dict
+            More options for the download
+        """
+        if not irods_path.is_collection():
+            raise ValueError("irods_path must be a collection.")
+
+        # get all data objects
+        coll = self.get_collection(irods_path)
+        all_objs = self._get_data_objects(coll)
+
+        for o in all_objs:
+            # ensure local folder exists
+            dest = Path(local_path, o[0].removeprefix(str(irods_path)).lstrip('/'))
+            if not dest.is_dir():
+                os.makedirs(dest)
+            self.irods_get(IrodsPath(self.session, o[0], o[1]), local_path, overwrite, options)
+
+    def get_size(self,
+                 item: Union[irods.data_object.iRODSDataObject, irods.collection.iRODSCollection]) -> int:
+        """Collect the sizes of a data object or a
+        collection.
+
+        Parameters
+        ----------
+        item : iRODSDataObject or iRODSCollection
+
+
+        Returns
+        -------
+        int
+            Total size [bytes] of the iRODS object or all iRODS objects in the collection.
+
+        """
+        if self.has_type_dataobj(item):
```

Comment:
Because of the return-statements, you don't really need the `elif` (just `if`), and `else`.

Reply:
Done. Just in case someone wants more information about the exception, we can add it again later.